tests/test_barman_cloud_backup_delete.py

```diff
@@ -141,23 +190,9 @@ def _create_backup_metadata(
                     }
                 )
             backup_metadata[backup_id]["files"] = mock_backup_files
-            backup_info = mock.MagicMock(name="backup_info", snapshots_info=None)
-            backup_info.backup_id = backup_id
-            if backup_name:
-                backup_info.backup_name = backup_name
-            backup_info.status = "DONE"
-            backup_info.end_time = datetime.datetime.strptime(
-                backup_id, "%Y%m%dT%H%M%S"
-            ) + datetime.timedelta(hours=1)
-            try:
-                backup_info.begin_wal = begin_wals[backup_id]
-                backup_info.timeline = int(backup_info.begin_wal[:8])
-            except KeyError:
-                pass
-            try:
-                backup_info.end_wal = end_wals[backup_id]
-            except KeyError:
-                pass
+            backup_info = self._create_backup_info(
+                backup_id, begin_wals, end_wals, backup_name
```

Comment:
```suggestion
                backup_id, begin_wals.get(backup_id), end_wals.get(backup_id), backup_name
```

That logic which replaces `begin_wals` and `end_wals` that were `None` with an empty dict (`{}`) would need to be performed in the beginning of this function.